.github/workflows/change-detection.yml

```diff
@@ -0,0 +1,55 @@
+name: Change detection
+
+on:
+  workflow_call:
+    outputs:
+      SCOPE:
+        description: "scope collection"
+        value: ${{ jobs.run.outputs.SCOPE }}
+      HAS_SCOPE:
+        description: "has scope"
+        value: ${{ jobs.run.outputs.HAS_SCOPE }}
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    outputs:
+      SCOPE: ${{ steps.set-scope.outputs.SCOPE }}
+      HAS_SCOPE: ${{ steps.set-scope.outputs.HAS_SCOPE }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Prepare environment
+        uses: ./.github/actions/prepare-environment
+
+      - name: Set scope
+        id: set-scope
+        shell: bash
+        run: |
+          # Get changed state   
+          lerna_ls=$(npx lerna ls --since=$(git merge-base --fork-point origin/dev))
+          
+          # Parse lerna ls output and print string like "b2c, web, temple, etc"
+          state=$(awk -F '-' '{if ($2 == "b2c" || $2 == "web" || $2=="ui" || $2=="temple") printf "%s%s", sep, $2; sep=","} END {print ""}' <<< $lerna_ls)
```

Comment:
@neretin-trike @kayman233 @Yeti-or 

Альтернативное решение на чистом bash от Васи, для получения значения типа ["web", "b2c"]

```bash
npx lerna ls --since origin/master | cut -d "-" -f 2 | xargs -I {} echo '\\\"{}\\\"' | tr '\n' ',' | sed 's/,$//' | xargs -I {} echo '[{}]'
```

Хочется понять какое нам лучше подойдет. Ваши доводы господа )

Reply:
да вроде этот вариант нормас